### DISCUSSION TITLE
Bienvenue sur le forum de discussion de QCMCam

### DISCUSSION THREAD

**Discussion by seb-cogez** [seb-cogez/QCMCam#1] [General]
## 👋 Bienvenue !
 Ce forum est un moyen de connecter les différents utilisateurs de QCMCam. Vous pouvez :
  * Poser vos questions
  * Partager vos idées
  * Soyez courtois et patients 💪.

S. COGEZ, développeur de QCMCam et MathsMentales

**Comment by loctogan** [comment]
Bonjour Sébastien,

Je suis tombé sur votre app qui l'air vraiment super, quel boulot!

Je viens de l'essayer à l'instant mais je constate un problème avec mon smartphone (apple), en effet la caméra utilisée pour scanner est la caméra en façade ce qui ne s'avère pas très pratique. 

J'ai cherché dans le code mais n'ai pas trouvé où modifier ce paramètre, pourriez-vous m'aider?

**Reply by seb-cogez** [reply]
Bonjour,
Pour les matériels apple, j'ai fait une petite vidéo sur ma chaine youtube. Je vous invite à aller voir cela.
cordialement

**Comment by ghost** [comment]
Bonjour Sébastien,

Professeure de mathématiques au collège, j'ai longtemps utilisé Plickers, c'est beau, pratique, mais ce n'est pas 100% libre.
Merci pour cette application qui répond à mes attentes, en permettant de sauvegarder des questionnaires et les partager entre collègues !
Je l'ai utilisée en classe, et cela fonctionne bien, même avec un smartphone, sans grande latence comme j'ai pu avoir avec l'autre application.

J'ai cependant deux questions :

**1°)** Lorsque les élèves ont répondu, on peut afficher la bonne réponse, mais on voit alors la réponse des élèves avec une couleur qui indique s'ils se sont trompés ou non. Est-il possible de ne pas afficher cela ? Si oui, comment ? (le but est d'éviter que les élèves qui se sont trompés se sentent mal)
J'ai essayé de cacher le nom des participants, mais le problème reste le même : le jeu des élèves est de retrouver qui est le N°X qui n'a pas eu la bonne réponse... 
J'ai essayé de montrer la répartition des réponses (et ne plus montrer les élèves) mais lorsqu'on passe à la question suivante, la répartition est affichée avec la ou les bonne(s) réponse(s). Les élèves n'ont donc pas à réfléchir, seulement à guetter la réponse.

**2°)** L'affichage en plein écran ne fonctionne pas, ou du moins, lorsque je clique sur le bouton (sur l'écran d'ordinateur ou sur le téléphone) cela ne change absolument rien.

Merci d'avance pour vos réponses, et bravo encore pour cet outil !

Camille

**Reply by seb-cogez** [reply]
Bonjour Camille,
1) La technique consiste (pour le moment), à montrer les stats en fin de scan, puis indiquer la bonne réponse. Lorsqu'on passe ensuite à la question suivante, l'affichage repasse sur la liste des participants. En faisant le test, je vois qu'il  y a un temps de latence où les résultats précédents sont affichés. Je vais changer cela rapidement.
2) C'est souvent des problèmes d'autorisation qui n'ont pas été données à la première tentative. Sur le smartphone, une fois la page ouverte, aller dans les paramètres de page et chercher un peu. Pour le navigateur, peut-être en essayer un autre, il faudrait être plus spécifique pour que je puisse vous aider un peu mieux.
Cordialement

**Reply by seb-cogez** [reply]
1. L'affichage des réponses lors d'un changement de question après avoir affiché les stats ne se fait plus. Il faudra forcer le rechargement de QCMCam pour voir les changements. (CTRL + F5)
Par ailleurs le cache qui disparaissait lors de l'association du smartphone ne le fera plus, il faudra commencer le scan pour que cela soit le cas, ou cliquer sur le cache pour qu'il disparaisse (le smartphone risque de garder une vieille version en mémoire)

**Reply by ghost** [reply]
Bonjour,

Merci pour cette réponse rapide !

Cela fonctionne parfaitement pour la question 1, merci. Je me permets de faire la remarque que cela fonctionne uniquement sur le smartphone (avec l'application) mais pas lorsque l'on fait la manipulation depuis l'ordinateur (via le site), pour les utilisateurs de webcam.
Si, depuis l'ordinateur, on clique sur la répartition des réponses, puis sur l'affichage de la bonne réponse, on obtient le graphique avec en vert la (ou les) bonne(s) réponse(s). En cliquant sur la question suivante, on voit non pas s'afficher le nom des participants, mais la répartition des réponses à cette nouvelle question (donc 0 réponse à chaque réponse) et en vert la (ou les) bonne(s) réponse(s).

Pour la question 2, j'utilise Google Chrome et j'ai essayé Microsoft Edge (l'établissement ne possède pas d'autre navigateur). L'affichage de base permet de voir sur la moitié gauche de l'écran les questions et sur la moitié droite la liste des participants ou la répartition des questions. 
Lorsque je clique sur le bouton plein écran ou fais ALT+Z sur le site, ou appuie sur le bouton de l'application, l'écran reste dans cette configuration, sans aucun changement.

**Comment by ghost** [comment]
Bonjour Sebastien,

J'ai voulu utiliser QcmCam ce matin avec mes élèves mais le message d'erreur suivant est apparu (deux navigateurs Chrome et Edge) :
**Reason : The SSL Server certificate is expired or not yet valid
Common name : qcmcam.net**

Impossible de forcer l'accès au site sur le réseau du collège. J'ai réussi à me débrouiller autrement, mais les élèves étaient super déçus de ne pas pouvoir faire "le jeu des codes" !

J'ai réussi à accéder au site depuis mon téléphone et mon ordinateur personnel, mais un message d'erreur était présent : 
**Votre connexion n'est pas privée
Des individus malveillants tentent peut-être de subtiliser vos informations personnelles sur le site qcmcam.net (mots de passe, messages ou numéros de carte de crédit, par exemple).
NET::ERR_CERT_DATE_INVALID**

En cliquant sur les paramètres avancés, voici ce qui s'affiche : 
**Impossible de vérifier que ce serveur est bien qcmcam.net, car son certificat de sécurité est arrivé à expiration au cours des dernières 24 heures. Cela peut être dû à une mauvaise configuration ou à l'interception de votre connexion par un pirate informatique. L'horloge de votre ordinateur indique actuellement jeudi 27 janvier 2022. Cela vous semble-t-il correct ? Si ce n'est pas le cas, vous devez corriger l'horloge de votre système, puis actualiser la page.**

On peut alors cliquer sur "Continuer vers le site qcmcam.net (dangereux)" et accéder au site.

Le référent TICE du collège m'a dit que le site n'était peut-être plus RGPD, mais n'avait pas d'autre idée.

En espérant que ce problème n'est pas trop sérieux et sera facile à résoudre.

Camille

**Reply by seb-cogez** [reply]
Merci pour le retour,
C'était un problème avec le renouvellement de certificat qui est géré normalement automatiquement tous les mois et a foiré ce matin apparemment. J'espère que cela n'aura pas lieu trop souvent.
C'est réparé.
